### DISCUSSION TITLE
It would be amazing to split the country data into regions

### DISCUSSION THREAD

**Discussion by christopherpickering** [umami-software/umami#1073] [Ideas]
Hey @mikecao , I know how you feel about analytics at the city level, but would you consider doing more subregions than just countries? Like states/provinces in North America?

North America, Russia, China, Australia and India cover huge chunks of the map. It would be really nice to have the data points reduced more to the size of France..

This would be especially handy in Canada where the language is different in different provinces. 

![image](https://user-images.githubusercontent.com/17788706/155390462-45258378-803d-4c0a-b61c-4f560e3c40ea.png)

It looks like this data (IP by region) isn't existing in the geolite db... but I wonder if there is another way to get it without too much trouble.

Region vs country doesn't seem to change the level of privacy of the users either.

**Comment by deshetti** [comment]
@christopherpickering You beat me to add this issue. This was one of the most requested features on Plausible and they added it just last month; it is pretty neat.

@mikecao You might be already aware of it, but some of the discussions around the same can be found here:

- https://github.com/plausible/analytics/discussions/102
- https://github.com/plausible/analytics/discussions/219

They moved to use https://db-ip.com/ from Geolite to support this. 

On a separate note, I installed this a few hours ago in our production for one of the sites and the countries are not populating as expected. I opened it as a separate issue: https://github.com/mikecao/umami/issues/984

**Reply by zx900930** [reply]
But the problem is dp-ip.com's api collecting user data via GA, and it doesn't like GDPR compliance(Which is one of the main benefits switching to umami).
Moving from Geolite to [GeoLite2](https://dev.maxmind.com/geoip/docs/databases/city-and-country/?lang=en#binary-databases) is a better way to go.

**Comment by christopherpickering** [comment]
Nice, well the other geo2 database does include the region.

**Comment by JezielMoura** [comment]
I met umami recently and found it to be an amazing tool, previously I was using Matomo. I'm from Brazil and my clients are all from tourism, so for them to know the region that is getting the most hits is essential for advertising. I believe there are more people like me, this role would be very welcome. If I can help with the development, I'm available.

**Comment by sankyutang** [comment]
I alos have the same issue in china

**Comment by ghost** [comment]
It would be great if, as an end user / self hosted server, I had the ability to setup the system to be more in depth with the geolocation tracking.
I have only had Umami running for about a week, but so far, I really like it.
I love the fact that Umami doesn't require Docker. (Looking at you Plausible!)
The only downside is the location tracking is way too vague.
Please add the ability to track further than country/region.
